cds_ils/importer/documents/api.py

```diff
@@ -0,0 +1,128 @@
+    hits_total = result.hits.total.value
+
+    if hits_total < 1:
+
+        click.secho(
+            "no document found with {} {}".format(search_term_name,
+                                                  search_term),
+            fg="red",
+        )
+        if raise_on_empty:
+            raise DocumentImportError(
+                "no document found with {} {}".format(search_term_name,
+                                                      search_term)
+            )
+    elif hits_total > 1:
+        click.secho(
+            "found more than one document with {} {}".format(search_term_name,
+                                                             search_term),
+            fg="red",
+        )
+        if raise_on_many:
+            raise DocumentImportError(
+                "found more than one document with {} {}".format(
+                    search_term_name, search_term)
+            )
+    return hits_total
+
+
+def search_documents_by_isbn(isbn):
+    """Find document by ISBN."""
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", identifiers__scheme="ISBN"),
+            Q("match", identifiers__value=isbn),
+        ],
+    )
+    return search.execute()
+
+
+def search_documents_by_doi(doi):
+    """Find document by ISBN."""
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", identifiers__scheme="DOI"),
+            Q("match", identifiers__value=doi),
+        ],
+    )
+    return search.execute()
+
+
+def get_document_by_exact_title_authors(title, authors):
+    """Find document by title and authors."""
+
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", title=title),
+            Q("terms", authors__full_name=authors),
+        ],
+    )
+
+    result = search.execute()
+    if len(result) > 0:
+        return result
+
+
+def fuzzy_search_document(title, authors):
+    build_fuzzy_authors_query = [Q({"fuzzy":
+        {
+            "authors.full_name": {
+                "value": author,
+                "fuzziness": "AUTO",
+                "prefix_length": 0,
+                "transpositions": True,
+                "max_expansions": 50
+            }
+        }
+    }) for author in authors]
+
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q({"fuzzy":
+                {
+                    "title": {
+                        "value": title,
+                        "fuzziness": "AUTO",
+                        "prefix_length": 0,
+                        "transpositions": True,
+                        "max_expansions": 50
+                    },
+                }
+            }),
+            Bool(
+                "bool",
+                should=build_fuzzy_authors_query
+            )
+        ],
+    )
+    result = search.execute()
+    if len(result) > 0:
+        return [x.pid for x in result.hits]
```

Comment:
you are going to get only the first 10/15...
In general, I would suggest to always return `search.execute()`, or even only `search` so you can choose to use `execute()` or `scan()` for example when used.
These are APIs methods, they should be slightly more generic without knowing who is calling them.